test/drivers/test_revd2.cc

```diff
@@ -14,69 +14,94 @@ class TestREVD2 : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// General test for RSVD:
-    /// Computes the decomposition factors, then checks A-U\Sigma\transpose{V}.
-    template <typename T, typename RNG>
-    static void test_REVD2_general(int64_t m, int64_t k, int64_t k_start, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state, int rank_expectation, T err_expectation) {
+    template <typename T>
+    struct REVD2TestData {
+        std::vector<T> A_buf;
+        std::vector<T> A;
+        std::vector<T> A_cpy;
```

Comment:
What's the difference between ``A_buf`` and ``A_cpy``?

Reply:
If ``A_buf`` is temporary workspace that's only used in one place then I don't think you need to allocate it in advance. Just allocate, use it to define ``A``, then deallocate. (Of course since this is std::vector you don't need explicit deallocation but you get what I mean.)

Reply:
I get that, will fix.
But this technically gets into the realm of reusing space vs readability.

Reply:
Okay nvm I see why that doesn't work well.

Please just use a different variable name here.

Reply:
> I get that, will fix.
But this technically gets into the realm of reusing space vs readability.

If you need to reuse space then it's better to use a more open-ended variable name like ``work`` or ``work1``. That way the person reading the code knows "Ah, this might be used for a few different purposes. Better be careful when messing with it."

Reply:
I actually switched it to A_cpy. Feel like its ok in this case, but at some point we'll need to decide on the space reusal stuff.